src/deriver/api.py

```diff
@@ -378,10 +378,19 @@ def derive_smiles_gb(self, n_children: int = 100, mut_rate: float = 0.01):
             filter_params = None
 
         for _ in range(n_children):
-            parent_a_smiles, parent_b_smiles = random.sample(self.data.seed_smiles, 2)
-            parent_a, parent_b = [Chem.MolFromSmiles(s) for s in (parent_a_smiles, parent_b_smiles)]
             try:
-                new_child = crossover_gb(parent_a, parent_b)
+                parent_a_smiles, parent_b_smiles = random.sample(self.data.seed_smiles, 2)
+                do_crossover = True
+                new_child = None
+            except ValueError:
+                do_crossover = False
+                assert len(self.data.seed_smiles) > 0
```

Comment:
better to assert this above, it has to be true in any case

Reply:
Crap, had already done this forthe other method